docs/docs/spec.md

```diff
@@ -452,6 +459,10 @@ condition is met, skip the remaining conditions.
 -   If the type usage is a component of a return type in an annnotation
     interface, its nullness operator is `MINUS_NULL`.
 
+-   TODO: Why does this only list two of the intrinsically non-null locations listed earlier?
```

Comment:
That is a good point.

I don't think I'd thought about this in much detail, but I suspect that my thought process was effectively this:

- For some of those locations, we don't need to care, since we [haven't made much attempt to address implementation code](https://github.com/jspecify/jspecify/issues/661). (This would apply to exception parameters and local variables, for example.)

- For some others, there's clearly no rule about nullness that would apply to them. (For example, what would it even mean to write `class Foo extends @Nullable Object`?)

- That leaves us with some cases that _look_ like fields and _look_ like methods but are actually _special_ fields (enum constants) and _special_ methods (annotation elements).

Given that, I was drawn to covering that final kind of case in the spec.

But I think that's wrong:

> This section defines how to determine the augmented types of most type usages in source code or bytecode where JSpecify nullness annotations are recognized.

An enum constant and a return type in an annotation interface\[*\] are both unrecognized locations. So even if we want to have rules like this, they don't belong here.

And then, once we do have them, there's not a clear reason to cover them but not cover other locations. The closest thing I have to a reason is the "criteria" in the bullet points above. Even those points would leave us open to questions about whether cases like receiver parameters meet the criteria.

I see at least two courses of action:

1. Just remove these rules. (We could then consider adding a non-normative comment in this section that refers back to the intrinsically non-nullable types mentioned ealier, perhaps with particular emphasis on the enum and annotation cases on account of my third bullet above.)

2. Split this section in two, with the first half covering recognized locations and the second half covering (for now, just these two) unrecognized locations.

Do you have a preference or another idea? The obvious other option would be to try to cover _all_ locations, but I'd at least like to declare that part of 1.0.1 or whatever we call the next version of the spec.

\[*\] I was having trouble finding that, it turns out because of the typo that I'll fix in https://github.com/jspecify/jspecify/pull/697 :)

Reply:
Hmm, enum constants may be a special case: We don't currently say that annotations are unrecognized on _the type of_ an enum constant, since there's no way to write them there in source code. But there _is_ a type in bytecode, as we note in https://jspecify.dev/docs/spec/#augmented-type-of-usage. So it would seem that either we need to say that annotations are unrecognized there (so that the normal rules for computing the type don't apply) or that we need to keep the rule for computing the nullness operator there (for the same reason).

Reply:
I've added one rule that covers all intrinsically non-null locations and calls out two examples.
What do you think?